### PR TITLE
App Sec Custom Rules

### PR DESCRIPTION
**Pull request by sarina-dd**
### What does this PR do?
Adds the Application Security Custom Rule doc

### Motivation
Jira request / OKR1

### Preview
https://docs-staging.datadoghq.com/sarina/app-sec-custom-rules/security_platform/application_security/custom_rules

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/security_platform/application_security/custom_rules.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+#### Joining queries
+
+Joining queries to span a timeframe can increase the confidence or severity of the Security Signal. For example, to detect a successful attack, both successful and unsuccessful tiggers can be correlated for a service.
+
+Queries are correlated together by using a `group by` value. The `group by` value is typically an entity (for example, `IP address` or `Service`), but can be any attribute. In this instance, joined queries technically hold same attribute value because the value must be the same for the case to be met. If a `group by` value doesn’t exist, the case will never be met. A Security Signal is generated for each unique `group by` value when a case is matched.
+
+For example, create a query that searches for the same `Service` activity, but append opposing HTTP status code queries:
+
+Query `successful_attack_trigger`: `security_activity_to_monitor` ``-@http_status_code:[500-599]`.
```
Comment (sarina-dd) [inline content/en/security_platform/application_security/custom_rules.md:55]:
```suggestion
Query `successful_attack_trigger`: `security_activity_to_monitor -@http_status_code:[500-599]`.
```

File: content/en/security_platform/application_security/custom_rules.md [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+
+Query `failed_attack_trigger`: `security_activity_to_monitor``@http_status_code:[500-599]`.
```
Comment (sarina-dd) [inline content/en/security_platform/application_security/custom_rules.md:57]:
```suggestion
Query `failed_attack_trigger`: `security_activity_to_monitor @http_status_code:[500-599]`.
```

File: content/en/security_platform/application_security/custom_rules.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+{{< /tabs >}}
+
+## Set a rule case
+
+{{< tabs >}}
+{{% tab "Threshold" %}}
+
+### Trigger
+
+Rule cases, such as `succesful trigger > 0`, are evaluated as case statements. Thus, the first case to match generates the signal. Create one or multiple rule cases, and click on the grey area next to them to drag and manipulate their ordering.
```
Comment (sarina-dd) [inline content/en/security_platform/application_security/custom_rules.md:81]:
```suggestion
Rule cases, such as `successful trigger > 0`, are evaluated as case statements. Thus, the first case to match generates the signal. Create one or multiple rule cases, and click on the grey area next to them to drag and manipulate their ordering.
```

File: content/en/security_platform/application_security/custom_rules.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
+---
+title: Writing Custom Detection Rules
+kind: documentation
+---
+
+## Overview
+
+To create a custom detection rule in Datadog, hover over **Security**, select **Detection Rules**, and select the **New Rule** button in the top right corner of the page.
```
Comment (sarina-dd) [inline content/en/security_platform/application_security/custom_rules.md:8]: Premise of custom rules goes here

Comment (sarina-dd) [reply content/en/security_platform/application_security/custom_rules.md:8]: Main example goes here

Comment (sarina-dd) [reply content/en/security_platform/application_security/custom_rules.md:8]: Primary use case: customizing OOTB rules

File: content/en/security_platform/application_security/custom_rules.md [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+
+**Note**: The `evaluation window` must be less than or equal to the `keep alive` and `maximum signal duration`.
+
+## Say what's happening
+
+The **Rule name** section allows you to configure the rule name that appears in the rules list view, as well as the title of the signal.
+
+The notification box has the same Markdown and preview features.
+
+### Template variables
```
Comment (sarina-dd) [inline content/en/security_platform/application_security/custom_rules.md:82]:
```suggestion
#### Template variables
```

**Comment by sarina-dd** [comment]
@arnaudbreton 👋 Hi there, made a few edits to the template and organized based on the app sec queries. Please take a look and let me know of any additional edits we need to make. I'm OOO tomorrow, so I'll check back in with you on Monday!

File: content/en/security_platform/application_security/custom_rules.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+- link: "/security_platform/application_security/troubleshooting"
+  tag: "Documentation"
+  text: "Troubleshoot common Datadog Application Security issues"
+---
+
+## Overview
+
+Application Security comes with a set of [out-of-the-box detection rules][1] which aim to catch attack attempts and vulnerability triggers that impact your production systems.
+
+However, there are situations where you may want to customize a rule based on your environment. For example, you may want to customize a rule that catches attack attempts on a service’s route. For instance, a pre-production development route accepting SQL and returning the results. Catching SQL attempts is noisy as the route is restricted to internal developers, therefore you can customize this rule to exclude these patterns.
```
Comment (arnaudbreton) [inline content/en/security_platform/application_security/custom_rules.md:17]: However, there are situations where you may want to customize a **detection** rule...

File: content/en/security_platform/application_security/custom_rules.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+In these situations, a custom detection rule can be created to exclude such events. This guide shows you how to create a custom detection rule for Application Security.
+
+## Configuration
+
+To customize an OOTB detection rule, you must first clone an existing rule. Navigate to your [Detection Rules][2] and select a rule. Scroll to the bottom of the rule and click the Clone Rule button. This now enables you to edit the existing rule.
+
+### Define AppSec query
+
+Construct an Application Security query. For example, create a query to monitor an endpoint for SQL injection attempts: `@appsec.type:sql_injection @http.url_details.path:"/debug-endpoint-executing-sql" env:production`.
```
Comment (arnaudbreton) [inline content/en/security_platform/application_security/custom_rules.md:29]: Sorry that's my bad, I thought the query would be used in the context of an Exclusion Filter. If it is a detection rule query, then it misses an `-`: `@appsec.type:sql_injection -@http.url_details.path:"/debug-endpoint-executing-sql" env:production`

**Review by arnaudbreton — CHANGES_REQUESTED** [review]

File: content/en/security_platform/application_security/custom_rules.md (hunk 6 above)
Comment (sarina-dd) [inline content/en/security_platform/application_security/custom_rules.md:17]:
```suggestion
However, there are situations where you may want to customize a rule based on your environment. For example, you may want to customize a detection rule that catches attack attempts on a service’s route. For instance, a pre-production development route accepting SQL and returning the results. Catching SQL attempts is noisy as the route is restricted to internal developers, therefore you can customize this rule to exclude these patterns.
```

File: content/en/security_platform/application_security/custom_rules.md (hunk 7 above)
Comment (sarina-dd) [inline content/en/security_platform/application_security/custom_rules.md:29]:
```suggestion
Construct an Application Security query. For example, create a query to monitor an endpoint for SQL injection attempts: `@appsec.type:sql_injection -@http.url_details.path:"/debug-endpoint-executing-sql" env:production`.
```

File: content/en/security_platform/application_security/custom_rules.md [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+In the “Notify” section, configure zero or more [notification targets][3] for each rule case.
+
+You can also create [notification rules][4] to eleviate manual edits to notification preferences for individual detection rules.
+
+### Time windows
+
+An `evaluation window` is specified to match when at least one of the cases matches true. This is a sliding window and evaluates in real-time.
+
+Once a signal is generated, the signal will remain “open” if a case is matched at least once within this `keep alive` window. Each time a new event matches any of the cases, the *last updated* timestamp is updated for the signal.
```
Comment (cswatt) [inline content/en/security_platform/application_security/custom_rules.md:77]:
```suggestion
When a signal is generated, it remains "open" if a case is matched at least once within this `keep alive` window. Each time a new event matches any of the cases, the *last updated* timestamp is updated for the signal.
```

File: content/en/security_platform/application_security/custom_rules.md [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+
+A signal will “close” regardless of the query being matched once the time exceeds the `maximum signal duration`. This time is calculated from the first seen timestamp.
```
Comment (cswatt) [inline content/en/security_platform/application_security/custom_rules.md:79]:
```suggestion
A signal will "close" regardless of the query being matched once the time exceeds the `maximum signal duration`. This time is calculated from the first seen timestamp.
```

File: content/en/security_platform/application_security/custom_rules.md [hunk 10]
```diff
@@ -0,0 +1,169 @@
...
+
+### Say what's happening
+
+The **Rule name** section allows you to configure the rule name that appears in the rules list view, as well as the title of the signal.
+
+The notification box has the same Markdown and preview features.
+
+#### Template variables
+
+Detection rules support template variables within the markdown notification box. Template variables permit injection of dynamic context from traces directly into a security signal and its associated notifications.
```
Comment (cswatt) [inline content/en/security_platform/application_security/custom_rules.md:93]:
```suggestion
Detection rules support template variables within the Markdown notification box. Template variables permit injection of dynamic context from traces directly into a security signal and its associated notifications.
```

File: content/en/security_platform/application_security/custom_rules.md [hunk 11]
```diff
@@ -0,0 +1,169 @@
...
+    "first_seen":"2022-01-26T13:23:33.000Z",
+    "last_seen":"2022-01-27T04:01:57.000Z"
+  },
+  "groupByPaths":[
+    "service"
+  ]
+}
+```
+
+For this attribute, you would use the following attribute in the “say what’s happening” section:
```
Comment (cswatt) [inline content/en/security_platform/application_security/custom_rules.md:134]:
```suggestion
For this attribute, use the following in the “say what’s happening” section:
```

**Review by cswatt — CHANGES_REQUESTED** [review]

File: content/en/security_platform/application_security/custom_rules.md [hunk 12]
```diff
@@ -0,0 +1,169 @@
...
+
+Provide a **name** for each rule case. This name is appended to the rule name when a signal is generated.
+
+#### Severity and notification
+
+Set the severity of the signal. The dropdown allows you to select an appropriate severity level (`INFO`, `LOW`, `MEDIUM`, `HIGH`, `CRITICAL`).
+
+In the “Notify” section, configure zero or more [notification targets][3] for each rule case.
+
+You can also create [notification rules][4] to eleviate manual edits to notification preferences for individual detection rules.
```
[the remark this answers is not in the document]
Comment (sarina-dd) [reply content/en/security_platform/application_security/custom_rules.md:71]: LOL, what a mix of evaluate + alleviate, but i'll stick with alleviate!

Comment (sarina-dd) [inline content/en/security_platform/application_security/custom_rules.md:71]:
```suggestion
You can also create [notification rules][4] to alleviate manual edits to notification preferences for individual detection rules.
```

**Review by cswatt — APPROVED** [review]

**Pull request merged**